### DISCUSSION TITLE
write CONNECT_TIMEOUT undefined:undefined

### DISCUSSION THREAD

**Discussion by VinothGox** [porsager/postgres#428] [Q&A]
Hi @porsager,
I am trying to connect psql database but I am getting this error "CONNECT_TIMEOUT undefined: undefined". I changed the connection_timeout value, but still, I got the error. Please let me know Why is this error coming?

**Comment by porsager** [comment]
Is it coming instantly? Which version?

It appears you're trying to connect to a host that is not responding, so check your hostname and port to ensure that it's correct.. Can you connect to it using `psql`?

**Comment by tx0c** [comment]
hit on this on AWS Lambda environment, did you guys reach some solution?

```json
{
  "errorType": "Error",
  "errorMessage": "write CONNECT_TIMEOUT undefined:undefined",
  "trace": [
    "Error: write CONNECT_TIMEOUT undefined:undefined",
    "    at connectTimedOut (file:///var/task/node_modules/postgres/src/connection.js:255:20)",
    "    at Timeout.done [as _onTimeout] (file:///var/task/node_modules/postgres/src/connection.js:1010:8)",
    "    at listOnTimeout (node:internal/timers:566:11)",
    "    at process.processTimers (node:internal/timers:507:7)",
    "    at new Query (file:///var/task/node_modules/postgres/src/query.js:35:9)",
    "    at sql (file:///var/task/node_modules/postgres/src/index.js:111:11)",
    "    at mycode (file:///var/task/lib/db.js:21:20)",
    "    at Runtime.handler (file:///var/task/handlers/......)",
    "    at Runtime.handleOnceNonStreaming (file:///var/runtime/index.mjs:1089:29)"
  ]
}
```

the target Postgres is in RDS here;

weirdly, if I change the sql query to something simple,  `SELECT 1+2` that works, so it should not be a connection issue, because connection is already made and it's working for some small data fetching from database;
e.g.: if I add a small number LIMIT that can query out the data also,
`SELECT id, title, summary, content FROM article WHERE ... LIMIT 3`

but a bigger number will fail with this CONNECT_TIMEOUT
`SELECT id, title, summary, content FROM article WHERE ... LIMIT 100`

and my original sql code is with parameter as the `${limit}`, and give it a `const limit = 3`, even that simple also got `write CONNECT_TIMEOUT`
`SELECT id, title, summary, content FROM article WHERE ... LIMIT ${limit}`

wonder @porsager have some way to show more where's the `undefined:undefined` from?
